core/src/toga/images.py

```diff
@@ -1,54 +1,90 @@
 from __future__ import annotations
 
+import warnings
+from io import BytesIO
 from pathlib import Path
+from warnings import warn
+
+try:
+    import PIL.Image as PIL_Image
+except ImportError:  # pragma: no cover
+    PIL_Image = None
 
 import toga
 from toga.platform import get_platform_factory
 
+# Make sure deprecation warnings are shown by default
+warnings.filterwarnings("default", category=DeprecationWarning)
+
 
 class Image:
     def __init__(
         self,
-        path: str | None | Path = None,
+        src: str | Path | bytes | PIL_Image.Image | None = None,
         *,
-        data: bytes | None = None,
+        path=None,  # DEPRECATED
+        data=None,  # DEPRECATED
     ):
         """Create a new image.
 
-        An image must be provided either a ``path`` or ``data``, but not both.
-
-        :param path: Path to the image to load. This can be specified as a string, or as
-            a :any:`pathlib.Path` object. The path can be an absolute file system path,
-            or a path relative to the module that defines your Toga application class.
-        :param data: A bytes object with the contents of an image in a supported format.
+        :param src: The source from which to load the image. Can be a file path
+            (relative or absolute, as a string or :external:class:`pathlib.Path`) or raw
+            binary data in any supported image format. Can also accept a
+            :external:class:`PIL.Image.Image` if Pillow is installed.
```

Comment:
Preference is to use `:any:`, rather than specific types like `:class:` - `:any:` "smart" resolves to the same URL, but it's shorter and doesn't require getting the type right every time you use it.